File: src/lib-common/pad-l.js
function pad_l(x,y,z)
{
 check(is_cool,x)

 if(is_uint(x))
 {
  const s=to_json(x)
  
  if(is_undef(y))
  {
   if(is_undef(z))
    return pad_l(s,"0",3)

   return pad_l(s,"0",z)
  }

  return pad_l(s,y,z)
 }

 check(is_str,x)
 check(is_str,y)
 check(is_uint,z)

 return x.padStart(z,y)
}

File: src/lib-common/pad-l.cs
fn pad_l x:cool padding length
 if is_uint x
  let s to_json x

  if is_undef padding
   if is_undef length
    ret pad_l s "0" 3

   ret pad_l s "0" length
  end

  ret pad_l s padding length
 end

 check is_str x
 check is_str padding
 check is_uint length

 ret x.padStart length padding
end
